src/emqtt.erl

```diff
@@ -1064,6 +1088,21 @@ terminate(Reason, _StateName, State = #state{conn_mod = ConnMod, socket = Socket
 code_change(_Vsn, State, Data, _Extra) ->
     {ok, State, Data}.
 
+-ifdef(UPGRADE_TEST_CHEAT).
+format_status(_, State) ->
+    [{data, [{"State", State}]},
+     {supervisor, [{"Callback", ?MODULE}]}].
+-else.
+%% gen_statm:format_status_default
+format_status(Opt, State_Data) ->
```

Comment:
In this case, where we don't have the cheat code enabled, would it make sense to delegate directly to `gen_statem:format_status/2`?

Reply:
we could omit this optional callback, actually.